### PR TITLE
provide fix for issue #7025

### PR DESCRIPTION
**Pull request by narration-sd**
This small change means consistent values will be returned for lstat modes on repositories 'symlinked' by composer require.  In truth these are linked by junctions when on Windows. 

With the prior statcacheclear according to file only, lstat() will return arbitrary values from other files or folders. With this general clear, php-windows will return a consistent zero, which although incorrect, suffices for Composer to determine it is indeed a junction, and use its correct handling.

The issues with php-windows lstat are known to the Composer team, and the serious problem from them in #7025 shows up at least on a fast multicore Windows machine, running php7.1-windows.  This patch appears to relieve this problem consistently.

### CODE REVIEW

File: src/Composer/Util/Filesystem.php [hunk 1]
```diff
@@ -663,8 +663,11 @@ public function isJunction($junction)
          * #define	_S_IFREG	0x8000
          *
          * Stat cache should be cleared before to avoid accidentally reading wrong information from previous installs.
+         
+         * n.b. now clearing the entire stat cache, for its troubles in php7.1-windows, appearing and
+         * diagnosed in composer github issue #7025 -- @narrationsd 22 Jan 2018
```
Comment (curry684) [inline src/Composer/Util/Filesystem.php]: Comments in open source are not "personal", if someone wants to know why someone did what they'll grab the blame/commit logs. I don't think the comment change is needed at all, the previous one still describes the resulting situation fine.

**Comment by Seldaek** [comment]
@curry684 do you concur that having *both* clearstatcache calls is necessary? Sounds a bit overkill but then again harmless.

**Comment by narration-sd** [comment]
Thanks, @curry684 , @Seldaek .  Jordi, the reason for having the two is that only with both does this pass both my tests and yours under CI.  It is completely crazy, yes.  But we are dealing with flawed underlying code. Take care, Clive.

**Comment by Seldaek** [comment]
Merged in 1.6/master as fd61a21bffb191a00980c64266ced43a820376e7 - thanks!

**Comment by curry684** [comment]
@Seldaek my verdict here is also 'better safe than sorry' with both safeguards. It's harmless and costs no performance, and given the consequences if it goes wrong I'm all for doing both.

The code will be retired anyway if we move to Win10 real symlinks.

**Comment by narration-sd** [comment]
err, the above test guarded with an isdir() check, of course; didn't quite say that...

**Pull request closed without merging**